### PR TITLE
[CTCP-1066] updated the endpoints and added validation for the request headers

### PR DESCRIPTION
**Pull request by prashanthmamidi**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala [hunk 1]
```diff
@@ -31,9 +31,45 @@ import javax.inject.Inject
...
 
-  def post(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+  def channelResponseGB(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
     request: Request[Source[ByteString, _]] =>
       request.body.runWith(Sink.ignore)
-      Accepted
+
+      validateHeaders(request.headers, "gb") match {
+        case true => Ok
+        case _    => BadRequest
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala]: Can we please add a message to the bad request to indicate what error is being thrown

File: app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala [hunk 2]
```diff
@@ -31,9 +31,45 @@ import javax.inject.Inject
...
+      }
   }
+
+  def validateHeaders(headers: Headers, officeCode: String): Boolean = {
+    var result = isHeaderExists("content-type", headers) &&
+      isHeaderExists("accept", headers) &&
+      headers.get("authorization").nonEmpty &&
+      headers.get("x-message-type").nonEmpty &&
+      headers.get("x-correlation-id").nonEmpty &&
+      headers.get("date").nonEmpty
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala]: Looks like we're missing the check for x-request-id

Comment (prashanthmamidi) [reply app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala]: I skipped this on purpose, as it's Optional as per the spec

**Comment by drnaylor** [comment]
Superseded by other PRs way back when, closing this.

**Pull request closed without merging**